airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/display_report.py

```diff
@@ -240,3 +543,75 @@ def _get_init_report_body(self, report_date: str, record_type: str, profile):
                 "tactic": tactic,
                 "metrics": ",".join(metrics_list),
             }
+
+
+class SponsoredDisplayV3ReportStream(ReportStream):
+    """
+    https://advertising.amazon.com/API/docs/en-us/sponsored-display/3-0/openapi#/Reports
+    """
+
+    API_VERSION = "reporting"  # v3
+    REPORT_DATE_FORMAT = "YYYY-MM-DD"
+    ad_product = "SPONSORED_DISPLAY"
+    report_is_created = HTTPStatus.OK
+    metrics_map = METRICS_MAP_V3
+    metrics_type_to_id_map = METRICS_TYPE_TO_ID_MAP
+
+    def __init__(self, config: Mapping[str, Any], profiles: List[Profile], authenticator: Oauth2Authenticator):
+        super().__init__(config, profiles, authenticator)
+        # using session without auth as API returns 400 bad request if Authorization header presents in request
+        # X-Amz-Algorithm and X-Amz-Signature query params already present in the url, that is enough to make proper request
+        self._report_download_session = requests.Session()
+
+    def report_init_endpoint(self, record_type: str) -> str:
+        return f"/{self.API_VERSION}/reports"
+
+    def _download_report(self, report_info: ReportInfo, url: str) -> List[dict]:
+        """
+        Download and parse report result
+        """
+        return super()._download_report(None, url)
+
+    def _get_init_report_body(self, report_date: str, record_type: str, profile):
+        metrics_list = self.metrics_map[record_type]
+
+        reportTypeId = "sdCampaigns"  # SponsoredDisplayCampaigns
+        group_by = ["campaign"]
+        filters = []
+
+        if record_type == "adGroups":
```

Comment:
Can we have a mapping of values {reportType: (reportTypeId, groupBy, filters)} instead of if...else, and parse it accordingly using get_body() method or similar?

Reply:
We _can_ do this, but to be honest, I don't want to waste time on this, since we [plan](https://github.com/airbytehq/airbyte-internal-issues/issues/10429) to migrate this connector to Async Low-code